src/pkg/plug/scanner/scan/controller.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scan
+
+import "github.com/goharbor/harbor/src/pkg/plug/scanner/dao/scan"
+
+// Options object for the scan action
+type Options struct{}
+
+// Option for scan action
+type Option interface {
+	// Apply option to the passing in options
+	Apply(options *Options) error
+}
```

Comment:
Maybe function will simpler than interface?  

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

https://github.com/goharbor/harbor/blob/master/src/core/middlewares/interceptor/quota/options.go#L25

Reply:
I'm totally fine with function options. Seems writing a function is much easier than implementing an interface.

Reply:
I'll change it in the implementation PR later. Will not do any changes to the current interface definitions.